btrfsmaintenance-functions

```diff
@@ -66,3 +66,29 @@ check_balance_running() {
 	fi
 	return 0
 }
+
+# function: wait_ac_power
+# parameter: {timeout} {AC power online device}
+#
+# wait until ac power goes online
+wait_ac_power() {
+	local timecount=0
+	local timeout=0
+	local ac_power_device=/sys/class/power_supply/AC/online
+
+	[ -n "$1" ] && [[ $1 =~ ^[0-9]+$ ]] && timeout=$1
+	[ -n "$2" ]&& ac_power_device="$2"
+
+	echo timeout $timeout ac_power_device $ac_power_device
+
+	if [ -f "${ac_power_device}" ]; then
+		while [ "$(cat ${ac_power_device})" -ne 1 ]; do
+			# AC is not online
+			[ $timeout -gt 0 ] && [ $timecount -ge $timeout ] && return 1
+			sleep 1s
+			timecount=$((timecount+1))
```

Comment:
Can we sleep for at least 30sec, so laptops' cpus can stay in deep sleep longer?  (or is it appropriate to make this configurable?)

Reply:
I'd rather make the timeout longer and not add another config variable.

Reply:
how long? I hate hidden timeouts :D